nix/flake/modules/joex.nix

```diff
@@ -0,0 +1,1728 @@
+overlay: { config, lib, pkgs, ... }:
```

Comment:
Key change: module by itself takes an overlay to later reference the package without a a call to callPackage.

The rest of changes to this file (`diff  ../module-joex.nix  modules/joex.nix`) are mostly `nixpkgs-fmt`.